pulp_rpm/app/advisory.py

```diff
@@ -123,23 +124,35 @@ def resolve_advisory_conflict(previous_advisory, added_advisory):
         previous_advisory.updated_date or previous_advisory.issued_date)
     added_updated_date = parse_datetime(
         added_advisory.updated_date or added_advisory.issued_date)
+    previous_updated_version = previous_advisory.version
+    added_updated_version = added_advisory.version
 
     previous_pkglist = set(previous_advisory.get_pkglist())
     added_pkglist = set(added_advisory.get_pkglist())
 
     if previous_updated_date == added_updated_date:
-        if not previous_pkglist.intersection(added_pkglist):
+        if previous_pkglist.intersection(added_pkglist):
```

Comment:
updated_dates_differ = previous_updated_date != added_updated_date
versions_differ = previous_updated_version != added_updated_version
pkglists_intersect = previous_pkglist.intersection(added_pkglist)

if (updated_dates_differ and not pkglists_intersect) or \
   (not updated_dates_differ and versions_differ and not pkglists_intersect):
     raise ...